hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java

```diff
@@ -499,6 +501,100 @@ public void findTargetByControllerIDWithDetails() {
                 .isEqualTo(set);
     }
 
+    @Test
+    @Description("Checks if an target can be found by an assigned distribution set")
+    public void shouldFindTargetByAssignedDistributionSetAndQuery() {
+        final DistributionSet set1 = testdataFactory.createDistributionSet("test1");
+        final DistributionSet set2 = testdataFactory.createDistributionSet("test2");
+        asList("1", "2", "3", "4", "5", "6", "7", "8", "9").forEach(this::createTargetWithAttributes);
+
+        final Page<Target> result1 = targetManagement.findByAssignedDistributionSetAndQuery(PAGE, set1.getId(),
+                "id==1 or id==2");
+
+        assertThat(result1.getTotalElements()).as("No target should be found when non is installed.").isEqualTo(0);
+
+        asList("1", "2", "3").forEach(id -> assignDistributionSet(set1.getId(), id));
+        asList("7", "8", "9").forEach(id -> assignDistributionSet(set2.getId(), id));
+
+        final Page<Target> result2 = targetManagement.findByAssignedDistributionSetAndQuery(PAGE, set1.getId(),
+                "id==1,id==2,id==9");
+
+        assertThat(result2.getTotalElements())
```

Comment:
Test would still pass if `result2` contains ID 1 and ID 3 by accident. Shouldn't you better check size and returned IDs as well?